### ISSUE TITLE
cryodrgn analyze fail

### ISSUE THREAD

**Issue by ts387** [zhonge/cryodrgn#50]
Hi Ellen,

My cryodrgn analyze run upon the first training (downsampled, small architecture) is failing with following error:

size mismatch for encoder.main.0.weight: copying a param with shape torch.Size([256, 16641]) from checkpoint, the shape in current model is torch.Size([256, 4])

I believe I took care to ensure -D and -Apix parameters corresponded to the original images when parsing poses and ctf.

I have made it work in the past and can't seem to figure out what I might be missing this time.

Please let me know if there's an obvious explanation. 

Many thanks!

Taha

**Comment by zhonge** [comment]
Can you provide the command you used to run `cryodrgn analyze` ?

**Comment by ts387** [comment]
Thank you! It was the following:

cryodrgn analyze 00_vae128_z8 49 --skip-umap --Apix 1.27265625

**Comment by zhonge** [comment]
The command looks fine. Can you also run `cryodrgn view_config [workdir]` and verify that the settings were the ones you originally used to train the model?

My guess is that a new cryodrgn run was created in the same workdir (but then cancelled), which might have overwritten the original configuration file.

**Comment by ts387** [comment]
Hi, still on 0.3.1 here (installed centrally) so don't have view_config as yet.

It has happened with each of my past few runs (separate dirs), which makes me wonder if I am doing something systematically wrong all of a sudden. I will try again in a fresh directory anyway and be extra cautious.

Is there something else I can give you that might help diagnose the issue? 

<img width="1680" alt="Error file" src="https://user-images.githubusercontent.com/79735016/115068580-57f2d500-9eea-11eb-8b08-ef135d62ebcd.png">

**Comment by zhonge** [comment]
Somehow there is a mismatch between the saved configuration and saved model. Do you have the original `cryodrgn train_vae` command you used?

Since you don't have the latest version, can you copy over this `view_config.py` script to your machine, and try it out with `python view_config.py [workdir]` 
https://github.com/zhonge/cryodrgn/blob/master/cryodrgn/commands/view_config.py

Then we can check if there is some kind of mismatch issue.

Thanks!
Ellen

**Comment by ts387** [comment]
Hey Ellen. Thanks so much. The original cryodrgn train_vae command was the following: 

cryodrgn train_vae particles128.mrcs --poses pose.pkl --ctf ctf.pkl --zdim 8 -n 50 --enc-mask -1 -o 00_vae128_z8 -v


The output from view_config is as follows:

{'dataset_args': {'ctf': '/home/ts387/cryodrgn/ctf.pkl',
                  'datadir': None,
                  'do_pose_sgd': False,
                  'ind': None,
                  'invert_data': True,
                  'keepreal': False,
                  'norm': [0, 131.9412],
                  'particles': '/home/ts387/cryodrgn/particles128.mrcs',
                  'poses': '/home/ts387/cryodrgn/pose.pkl',
                  'window': True},
 'lattice_args': {'D': 129, 'extent': 0.5, 'ignore_DC': True},
 'model_args': {'domain': 'fourier',
                'enc_mask': -1,
                'encode_mode': 'resid',
                'pdim': 256,
                'pe_dim': None,
                'pe_type': 'geom_lowf',
                'players': 3,
                'qdim': 256,
                'qlayers': 3,
                'zdim': 8},
 'seed': 27793}


-Taha

**Comment by zhonge** [comment]
I wasn't able to reproduce this error with your training settings. I suspect the saved `weights.49.pkl` got overwritten in your directory and so now there's a mismatch between the configuration file and the saved `weights.49.pkl`. Do you think this could have happened? Let me know if it's still broken in a fresh run in a new directory.

FYI: I would leave off the `--enc-mask -1` option. I should clarify that the "mask" is applied in Fourier space, so the default value effectively considers the whole image and will be more parameter efficient.

**Comment by ts387** [comment]
Hi Ellen. It went away following a fresh run in a new directory. I still don't know how or when the weights could have been overwritten, but seems like I'll be fine for now.

Thank you so much for your time :)!

Taha

**Comment by zhonge** [comment]
That's great to hear! Closing
